front/src/applications/operationalStudies/hooks/useScenarioData.ts

```diff
@@ -154,10 +160,26 @@ const useScenarioData = (scenario: ScenarioResponse, infra: InfraWithState) => {
     [projectedTrainsById]
   );
 
-  const timetableItemUsedForProjection = useMemo(
-    () => (trainIdUsedForProjection ? timetableItemsById.get(trainIdUsedForProjection) : undefined),
-    [trainIdUsedForProjection, timetableItems]
-  );
+  const pathUsedForProjection = useMemo(() => {
+    if (!trainIdUsedForProjection) return undefined;
+    if (!isOccurrenceId(trainIdUsedForProjection)) {
+      return timetableItemsById.get(trainIdUsedForProjection)?.path;
+    }
+
+    const pacedTrainId = extractPacedTrainIdFromOccurrenceId(trainIdUsedForProjection);
+    const pacedTrain = timetableItemsById.get(pacedTrainId);
+    if (!pacedTrain || !isPacedTrainResponseWithPacedTrainId(pacedTrain)) throw new Error();
```

Comment:
Maybe you could contextualize the error with a message to be more clearer

Reply:
Done :white_check_mark: